app/models/physical_storage.rb

```diff
@@ -46,6 +47,47 @@ def refresh_ems
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
+  def raw_delete_physical_storage
+    raise NotImplementedError, _("raw_delete_physical_storage must be implemented in a subclass")
+  end
+
+  def ensure_delete_valid
```

Comment:
You already are using `supports` so this extra check looks superfluous

Reply:
I wasn't familiar with the unsupported_reason concept so I tried to do something similar to raise an error msg. I changed it to use unsupported_reason like you suggested